extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/SmallRyeReactiveMessagingLifecycle.java

```diff
@@ -37,13 +38,22 @@ void onStaticInit(@Observes @Initialized(ApplicationScoped.class) Object event,
     }
 
     void onApplicationStart(@Observes @Priority(Interceptor.Priority.LIBRARY_BEFORE) StartupEvent event) {
+        // We do not want a request scope during the wiring, or it will be propagated and never terminated.
+        boolean isRequestScopeActive = Arc.container().requestContext().isActive();
+        if (isRequestScopeActive) {
+            Arc.container().requestContext().deactivate();
+        }
         try {
             mediatorManager.start();
         } catch (Exception e) {
             if (e instanceof DeploymentException || e instanceof DefinitionException) {
                 throw e;
             }
             throw new DeploymentException(e);
+        } finally {
+            if (isRequestScopeActive) {
+                Arc.container().requestContext().activate();
```

Comment:
I really don't know, but I'm wondering if the initial request context shouldn't be restored rather than a new one being activated? Are we ok to throw a way to context state?

Reply:
When you say it, it makes sense. I've done as you suggest, but I didn't resolve the conversation to double check

Reply:
I think this makes sense.